example/use_cases/employer_university_flow/main.go

```diff
@@ -0,0 +1,183 @@
+// They will request for the information, the student will respond.
+//
+// We use two different did methods here. did:key and a custom did method specified
+// in this file: did:example. The university uses did:example and the
+// user uses did:key.
+
+// InitalizationStep: Initialize the Wallet/Holder and the University
+// Step 0: Univesity issues a VC to the Holder and sends it over
+// Step 1: Verifier requests data from the holder
+// Step 2: Holder sends credential
+// Step 3: Verifier grants access based on the result
+
+//                          |--------------------------|
+//                          |                          |
+//                          |   Issuer (University)    |
+//                          |                          |
+//                          |__________________________|
+//                             /                       \
+//                            /                          \ Trusts University
+//      -----------------    / Issues VC               -------------------------
+//     |                |   /                         |                         |
+//     |   Holder       |  / <--------------------->  |    Verifier (Employer)  |
+//     |      \Wallet   |      PresentationRequest    |                         |
+//     |----------------|                              --------------------------
+//
+//  A couple nuances that are necessary to understand at a high level before
+//  digging into this code.
+//
+//  1. A DID can be used against different method types. Each method has
+//  different funtions. For example, bitcoin works differently than peer.
+//  did:btcn vs. did:peer is how these methods specified.
+//
+//  2. A Verified Credential (VC) contains a cyrptographic proof, either explicit
+//   or embedded into the VC. For the purposes of this demo, the proof is
+//   embedded in a JSON Web Token (JTW)
+//
+//  3. When the Verifier wants to validate a user, they send a Presentation Request.
+//   The response will contain the VC. The Verifier will be able to determine if the VC
+//   has been tampered with due to the proof.
+//
+//   The objects being created are in the following order:
+//
+//  1. DID's and wallets are created for the holder, issuer, and verifier
+//  3. VC is issued to the student holder
+//  4. PresentationRequest submitted by the verifier
+//  5. PresentationSubmission returned by the holder
+//  6. Authorization from the Verifier.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	util "github.com/TBD54566975/ssi-sdk/example"
+	emp "github.com/TBD54566975/ssi-sdk/example/use_cases/employer_university_flow/employer_university_flow"
+
+	"github.com/TBD54566975/ssi-sdk/credential/signing"
+	"github.com/TBD54566975/ssi-sdk/cryptosuite"
+	"github.com/TBD54566975/ssi-sdk/did"
+	"github.com/sirupsen/logrus"
+)
+
+var cw = emp.CustomStepWriter{}
+
+// Set to debug mode here
+var debug = os.Getenv("DEBUG")
+
+func init() {
+	if debug == "1" {
```

Comment:
can define modes as consts? perhaps even a type
`type Mode string`

Reply:
got it.